### DISCUSSION TITLE
Anzeige Display

### DISCUSSION THREAD

**Discussion by frank-tanne** [tbnobody/OpenDTU#1745] [General]
Seit den letzten Update ist die anzeige im Display hochgerutscht. Kann man mit Gehäuse schlecht ablesen. Könnte man das wieder rückgängig machen?

**Comment by MetaChuh** [comment]
@frank-tanne 

bei mir hat sich nichts verändert (v24.2.12)
ausreichend abstand zwischen grafik/text und oled displaykanten.
(siehe screenshot)

ist bei dir das display im gehäuse verrutscht ?

![IMG_5023](https://github.com/tbnobody/OpenDTU/assets/45773305/a8d3bdf5-e700-4d39-95e0-fc8786f1432f)

**Reply by frank-tanne** [reply]
Nein leider nicht.

**Comment by frank-tanne** [comment]
Alt
![20240214_113228](https://github.com/tbnobody/OpenDTU/assets/141310741/f0b4a204-3371-49f8-9a81-c64eded909d0)
Neu
![20240214_113214](https://github.com/tbnobody/OpenDTU/assets/141310741/3f989239-668f-499e-8d7e-6db414a3e43b)

**Comment by MetaChuh** [comment]
dein display sitzt im gehäuse zu weit oben.
unten sieht man am foto noch den schwarzen rand, wo es keine pixel mehr gibt, oben nicht.

am besten neu zentrieren, und gegebenfalls das gehäuse an der oberen displaykante etwas nachfeilen.

**Reply by Tiese** [reply]
Hallo @frank-tanne,
da hat @MetaChuh zwar recht, aber man sieht auch deutlich, dass beim 2. Bild der Zeilenabstand etwas größer ist!

**Reply by frank-tanne** [reply]
Das meine ich auch. Muss ich halt damit leben. Am Gehäuse kann ich nichts mehr machen das ist verklebt.

**Reply by MetaChuh** [reply]
> Am Gehäuse kann ich nichts mehr machen das ist verklebt.

oje.
ja, die zeilenabstände sind wegen der lesbarkeit wirklich vergrößert worden, hatte ich nicht negativ bemerkt, da bei mir das display bis zum rahmen frei und nicht verdeckt ist.

[make efficient use of available display area](https://github.com/tbnobody/OpenDTU/commit/25a66a17229747064344aae16ef098212ab01158)

```
fix calculation of the text baselines, using getAscent() in favor of
getMaxCharHeight(), which includes ascent and descent. this moves the
first text up and allows to insert margin between the lines until the
display area is fully utilized.

on large displays, if the small diagram is selected, keep the first line
rather low to avoid collision with the diagram y-axis label. in this mode,
there is still more space between the text lines as before, allowing for
improved readability.
```

**Comment by ms1958** [comment]
Hallo zusammen, Hin und wieder kommt es bei mir vor, das das Diplay invertiert ist. also Schwarze Schrift auf Hellem Grund.
ist das gewollt ? ich habe mittlerweile ein neues Display und einen neue ESP32 drin. Sowohl bei der alten, als auch bei der neuen Hardware kommt das gelegentlich vor. Nach einem Neustart ist wieder alles normal. 
( Gehäuse, das war mal eins von diesen China billig GridTie Invertern. Ist abgeraucht beim ersten versuch )
![IMG_20240214_143539 (Groß)](https://github.com/tbnobody/OpenDTU/assets/145383336/f788c8a0-8d77-47dd-aa94-951c2973b343)
![IMG_20240214_143542 (Groß)](https://github.com/tbnobody/OpenDTU/assets/145383336/86d93444-c522-42ed-9580-27ec29f8b2b8)

**Reply by frank-tanne** [reply]
Sieht ja nicht mal schlecht aus. Was ist denn das für Display größe?

**Reply by ms1958** [reply]
1,4" , gestern habe ich ein 2,5" Display (Aliexpress) bekommen:

**Reply by ms1958** [reply]
![IMG_20240214_182306 (Groß)](https://github.com/tbnobody/OpenDTU/assets/145383336/9c7d19fd-b188-44e8-8770-a1a5a80087ca)

**Reply by ms1958** [reply]
hier mal das 2,5" Display

**Reply by MetaChuh** [reply]
@ms1958 

optisch seeehr cool dein umgebautes gehäuse 👍

display invers weiß auf schwarz sieht auch irgendwie hammer aus.
glaub, nicht beabischtigt, aber wer weiß, vielleicht wird invers irgendwann mal ne absichtliche option, schlecht für oled, schön für die augen 😉

dein großes 2.5" display lacht mich auch definitiv mit "kauf miiich" an 😄

**Reply by ms1958** [reply]
sorry, nicht 2,5 inch sindern 2,42 inch
![2,42inch](https://github.com/tbnobody/OpenDTU/assets/145383336/11a1c337-3d90-48d5-9fc8-376ca4870868)

**Reply by ms1958** [reply]
das war das ehemalige gridtie Gehäuse mit dem ersten display in weiss.
Die Stehbolzen passten nach ein bischen feilen an der Platien, die haargenau da rein passte.
![dtu](https://github.com/tbnobody/OpenDTU/assets/145383336/d1a723c3-26f1-4743-9453-a7ccc9d16985)
da ist noch ein Mini DC/DC wandler von 12V auf 5V reingekommen, 
die ich von meinem WifI rs232 Gateway mit AccesPoint abgezwackt habe zur versorgung
